superset/daos/tag.py

```diff
@@ -51,14 +51,29 @@ def create_custom_tagged_objects(
         object_type: ObjectType, object_id: int, tag_names: list[str]
     ) -> None:
         tagged_objects = []
-        for name in tag_names:
+
+        # striping and de-dupping
+        clean_tag_names: set[str] = {tag.strip() for tag in tag_names}
+
+        for name in clean_tag_names:
             type_ = TagType.custom
-            tag_name = name.strip()
```

Comment:
This could have led to dups, if you add "hello" and "hello  " it would probably trigger an error. Wasn't the root cause but might as well clean it up.